Single Player Crash

forge-1.12.2-14.23.4.2705
ancientwarfare-1.12.2-2.6.1.296

Description: There was a severe problem during mod loading that has caused the game to fail

net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from Ancient Warfare NPCs (ancientwarfarenpc)
Caused by: java.lang.UnsupportedOperationException
	at com.sun.nio.zipfs.ZipPath.toFile(ZipPath.java:597)
	at net.shadowmage.ancientwarfare.npc.skin.NpcSkinManager.lambda$loadSkinsFromSource$0(NpcSkinManager.java:65)
	at net.shadowmage.ancientwarfare.npc.skin.NpcSkinManager$$Lambda$2261/1349790416.apply(Unknown Source)
	at net.shadowmage.ancientwarfare.core.util.FileUtils.findFiles(FileUtils.java:45)
	at net.shadowmage.ancientwarfare.npc.skin.NpcSkinManager.loadSkinsFromSource(NpcSkinManager.java:63)
	at net.shadowmage.ancientwarfare.npc.skin.NpcSkinManager.loadSkins(NpcSkinManager.java:56)
	at net.shadowmage.ancientwarfare.npc.proxy.NpcClientProxy.loadSkins(NpcClientProxy.java:115)
	at net.shadowmage.ancientwarfare.npc.AncientWarfareNPC.postInit(AncientWarfareNPC.java:115)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:627)
	at sun.reflect.GeneratedMethodAccessor2.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:218)
	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:196)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:135)
	at net.minecraftforge.fml.common.Loader.initializeMods(Loader.java:749)
	at net.minecraftforge.fml.client.FMLClientHandler.finishMinecraftLoading(FMLClientHandler.java:329)
	at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:534)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:377)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

https://pastebin.com/hYtEbMEA

--- a/src/main/java/net/shadowmage/ancientwarfare/npc/skin/NpcSkinManager.java
+++ b/src/main/java/net/shadowmage/ancientwarfare/npc/skin/NpcSkinManager.java
@@ -57,12 +57,13 @@ public class NpcSkinManager {
 		loadSkinsFromSource(new File(SKINS_CONFIG_PATH), "");
 	}
 
+	@SuppressWarnings("squid:S3725") //can't use toFile().exists() as it's unsupported in ZipPath
 	private void loadSkinsFromSource(File source, String base) {
 		HashMap<String, Set<String>> imageMap = new HashMap<>();
 
 		FileUtils.findFiles(source, base, root -> {
 			Path fPath = root.resolve("skin_pack.meta");
-			if (fPath != null && fPath.toFile().exists()) {
+			if (fPath != null && Files.exists(fPath)) {
 				try (BufferedReader reader = Files.newBufferedReader(fPath)) {
 					readImageMap(reader, imageMap);
 				}

